typescript/sdk/src/utils/MultiGeneric.ts

```diff
@@ -54,7 +54,7 @@ export class MultiGeneric<Value> {
     return Object.fromEntries(entries);
   }
 
-  remoteChains(name: ChainName): ChainName[] {
+  async remoteChains(name: ChainName): Promise<ChainName[]> {
     return this.chains().filter((key) => key !== name);
```

Comment:
this isnt async, why change here?

Reply:
this fetches from onchain, it's async

Reply:
Yorke is correct, this isn't async, only the HypErc20App child is. But you may need to make this parent async too for the class types to be valid, I assume that's why you changed it.

I believe the original intention of these MultiGeneric-based classes was to have the `App` children take all the configs up front at construction time so basic methods like this wouldn't need to be async. This new approach deviates from that but it's probably fine.  

This multi-generic concept of shaping a class around a known map of `ChainName` to a single `Value` makes sense in some cases but not in others. That's partly why I didn't make WarpCore a HypErc20App; its configs are more complex.

Reply:
this is nbd to me